### PR TITLE
TUSD Offboarding Plan Part II - Review

### PR DESCRIPTION
**Pull request by marczeller**
Tests are currently failing on Repay action.

### CODE REVIEW

File: src/20230925_AaveV2_Eth_TUSDOffboardingPlanPartII/AaveV2_Ethereum_TUSDOffboardingPlanPartII_20230925.sol [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    AaveV2Ethereum.COLLECTOR.transfer(
+      AaveV2EthereumAssets.BUSD_A_TOKEN,
+      address(this),
+      aBUSDBalance > availableBUSD ? availableBUSD : aBUSDBalance
+    );
+    AaveV2Ethereum.POOL.withdraw(
+      AaveV2EthereumAssets.BUSD_UNDERLYING,
+      type(uint256).max,
+      address(AaveV2Ethereum.COLLECTOR)
+    );
```
Comment (kyzia551) [inline src/20230925_AaveV2_Eth_TUSDOffboardingPlanPartII/AaveV2_Ethereum_TUSDOffboardingPlanPartII_20230925.sol:32]: I would recommend to deduplicate this code, and wrap it into function

File: src/20230925_AaveV2_Eth_TUSDOffboardingPlanPartII/AaveV2_Ethereum_TUSDOffboardingPlanPartII_20230925.sol [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+import {IV2RateStrategyFactory} from 'lib/aave-helpers/src/v2-config-engine/IV2RateStrategyFactory.sol';
+
+/**
+ * @title TUSD Offboarding Plan Part II
+ * @author Marc Zeller - Aave-Chan Initiative
+ * - Snapshot: https://snapshot.org/#/aave.eth/proposal/0x95cca29a9cdcaf51bb7331a9516d643a5c88f8ddce86c5f3920c2ae4d604193f
+ * - Discussion: https://governance.aave.com/t/arfc-tusd-offboarding-plan-part-ii/14863
+ */
+contract AaveV2_Ethereum_TUSDOffboardingPlanPartII_20230925 is AaveV2PayloadEthereum {
+  function _preExecute() internal override {
```
Comment (kyzia551) [inline src/20230925_AaveV2_Eth_TUSDOffboardingPlanPartII/AaveV2_Ethereum_TUSDOffboardingPlanPartII_20230925.sol:16]: Don't see LT update to 65% for TUSD

**Review by kyzia551 — CHANGES_REQUESTED** [review]

File: src/20230925_AaveV2_Eth_TUSDOffboardingPlanPartII/AaveV2_Ethereum_TUSDOffboardingPlanPartII_20230925.t.sol [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+    // check it's not bricked
+    ReserveConfig memory configBUSDAfter = _findReserveConfigBySymbol(allConfigsAfter, BUSD_SYMBOL);
+    _withdraw(
+      configBUSDAfter,
+      AaveV2Ethereum.POOL,
+      0xc579a79376148c4B17821C5Eb9434965f3a15C80,
+      1 ether
+    ); // aBUSD whale
+
+    // this test is failing: FAIL. Reason: _repay() : ERROR
```
Comment (kyzia551) [inline src/20230925_AaveV2_Eth_TUSDOffboardingPlanPartII/AaveV2_Ethereum_TUSDOffboardingPlanPartII_20230925.t.sol:70]: Could you please elaborate what this line means?

File: src/20230925_AaveV2_Eth_TUSDOffboardingPlanPartII/AaveV2_Ethereum_TUSDOffboardingPlanPartII_20230925.t.sol [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+    GovHelpers.executePayload(vm, BUSDPayload, AaveGovernanceV2.SHORT_EXECUTOR);
+
+    // check balances are correct
+    uint256 aBUSDBalanceAfter = IERC20(AaveV2EthereumAssets.BUSD_A_TOKEN).balanceOf(
+      address(AaveV2Ethereum.COLLECTOR)
+    );
+    uint256 BUSDBalanceAfter = IERC20(AaveV2EthereumAssets.BUSD_UNDERLYING).balanceOf(
+      address(AaveV2Ethereum.COLLECTOR)
+    );
+    assertApproxEqAbs(aBUSDBalanceAfter, 0, 2000 ether, 'aBUSD_LEFTOVER');
```
Comment (kyzia551) [inline src/20230925_AaveV2_Eth_TUSDOffboardingPlanPartII/AaveV2_Ethereum_TUSDOffboardingPlanPartII_20230925.t.sol:54]: why is here so big delta?

**Review by kyzia551 — CHANGES_REQUESTED** [review]

File: src/20230925_AaveV2_Eth_TUSDOffboardingPlanPartII/AaveV2_Ethereum_TUSDOffboardingPlanPartII_20230925.sol (hunk 2 above)
Comment (marczeller) [reply src/20230925_AaveV2_Eth_TUSDOffboardingPlanPartII/AaveV2_Ethereum_TUSDOffboardingPlanPartII_20230925.sol:20]: fixed

File: diffs/pre-BUSD-Payload-activation_20230804_post-BUSD-Payload-activation_20230804.md [hunk 5]
```diff
@@ -1,27 +1,21 @@
 ## Reserve changes
```
Comment (kyzia551) [inline diffs/pre-BUSD-Payload-activation_20230804_post-BUSD-Payload-activation_20230804.md:1]: Please remove changes in this file

**Review by kyzia551 — CHANGES_REQUESTED** [review]

File: src/20230925_AaveV2_Eth_TUSDOffboardingPlanPartII/AaveV2_Ethereum_TUSDOffboardingPlanPartII_20230925.sol (hunk 1 above)
Comment (marczeller) [reply src/20230925_AaveV2_Eth_TUSDOffboardingPlanPartII/AaveV2_Ethereum_TUSDOffboardingPlanPartII_20230925.sol]: fixed by @JosepBove

File: src/20230925_AaveV2_Eth_TUSDOffboardingPlanPartII/AaveV2_Ethereum_TUSDOffboardingPlanPartII_20230925.t.sol [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+  string public constant BUSD_SYMBOL = 'BUSD';
+  string public constant TUSD_SYMBOL = 'TUSD';
+
+  function setUp() public {
+    vm.createSelectFork(vm.rpcUrl('mainnet'), 18212146);
+  }
+
+  function testBUSD() public {
+    ReserveConfig[] memory allConfigsBefore = createConfigurationSnapshot(
+      'pre-BUSD-Payload-activation_20230804',
```
[the remark this answers is not in the document]
Comment (marczeller) [reply src/20230925_AaveV2_Eth_TUSDOffboardingPlanPartII/AaveV2_Ethereum_TUSDOffboardingPlanPartII_20230925.t.sol]: fixed

File: src/20230925_AaveV2_Eth_TUSDOffboardingPlanPartII/TUSDOffboardingPlanPartII.md [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+## Simple Summary
+
+This AIP proposes the continuation of the [TUSD offboarding plan](https://governance.aave.com/t/arfc-tusd-offboarding-plan/14008) from Aave V2 by adjusting its risk parameters, mirroring the approach taken in the "[ARFC] BUSD Offboarding Plan Part III".
+
+## Motivation
+
+This proposal is a continuation of stablecoins offboarding of the Aave V2 market.
+TUSD is proposed to adopt the "final stage" offboarding parameters to incentivize repayment and withdrawal of liquidity in Aave V2 Pool
+
+aBUSD will be withdrawn and converted to BUSD.
```
Comment (Zer0dot) [inline src/20230925_AaveV2_Eth_TUSDOffboardingPlanPartII/TUSDOffboardingPlanPartII.md:16]: Also converts TUSD so that should be specified

**Review by Zer0dot — APPROVED** [review]

**Pull request merged**